scanner/src/main/kotlin/ScanController.kt

```diff
@@ -212,7 +213,8 @@ class ScanController(
      * path. Packages without a resolved provenance are not included in the result.
```

Comment:
Please also mention the sorting in the function docs and ideally also explain the reason.

Reply:
Please review the wording.

Reply:
Nit: "amount of path separator**s**", comma after "For instance"
I would remove the last TODO sentence, the `Scanner` itself will always have to take packages and then do the provenance resolution. This function is related to finding packages which share the same provenance. After the provenance resolution this data is passed to scanners which operate either on a reference package like FossId or that operate on provenances like the current ScanCode implementation.